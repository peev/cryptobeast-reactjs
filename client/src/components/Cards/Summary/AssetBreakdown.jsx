// @flow
import React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Highcharts from 'highcharts';
import HighchartsReact from 'highcharts-react-official';
import { inject, observer } from 'mobx-react';

const styles = () => ({
  text: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    padding: '10px 23px',
    color: 'white',
    backgroundColor: '#133140',
    fontSize: '16px',
    fontWeight: '500',
    // textTransform: 'uppercase',
    height: '21px',
  },
  container: {
    height: '320px',
    paddingRight: '80px',
    paddingTop: '6px',
    boxShadow: 'inset 0 7.5px 9px -7px rgba(0,0,0,0.6)',
  },
  containerParagraph: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    paddingTop: '0 !important',
    textAlign: 'center',
  },
  paper: {
    marginLeft: '20px',
  },
});

type Props = {
  classes: Object,
  PortfolioStore: Object,
};

@inject('PortfolioStore')
@observer
class AssetBreakdown extends React.Component<Props> {
  state = {};

  render() {
    const { classes, PortfolioStore } = this.props;
    const options = {
      chart: {
        plotBackgroundColor: null,
        plotBorderWidth: null,
        plotShadow: false,
        type: 'pie',
        height: 320,
      },
      title: {
        text: null,
      },
      tooltip: {
        valueSuffix: '%',
      },
      plotOptions: {
        pie: {
          allowPointSelect: true,
          cursor: 'pointer',
          size: 230,
          center: [205, 115],
          dataLabels: {
            enabled: true,
          },
          showInLegend: true,
        },
      },
      credits: {
        enabled: false,
      },
      legend: {
        align: 'right',
        verticalAlign: 'middle',
        layout: 'vertical',
      },
      series: [{
        name: 'Asset Breakdown:',
        colorByPoint: true,
        data: PortfolioStore.summaryAssetsBreakdown,
      }],
    };


    return (
      <Paper className={classes.paper}>
        <Grid container >
          <Grid item xs={12} sm={12} md={12} className={classes.containerParagraph}>
            <Typography
              variant="title"
              id="modal-title"
              className={classes.text}
            >
              Asset Allocation
            </Typography>
          </Grid>

          <Grid item xs={12} sm={12} md={12} id="main">
          <div className={classes.container}>
            <HighchartsReact
              highcharts={Highcharts}
              options={options}
            />
          </div>
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

export default withStyles(styles)(AssetBreakdown);
