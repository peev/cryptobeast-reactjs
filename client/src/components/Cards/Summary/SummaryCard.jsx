// @flow
import React, { SyntheticEvent } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Popover from '@material-ui/core/Popover';
import Info from '@material-ui/icons/Info';
import statsCardStyle from './../../../variables/styles/statsCardStyle';

const styles = () => ({
  cardContainer: {
    width: '18%',
    minWidth: '150px',
    height: 'auto',
    overflow: 'unset',
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    '& path': {
      fill: '#133140',
    },
  },
  cardHeader: {
    position: 'relative',
    top: '-20px',
    left: '15px',
    height: '60px',
    width: '60px',
    backgroundColor: '#fff',
    border: '1px solid #133140',
    padding: '0',
  },
  icon: {
    color: 'white',
    fontSize: '35px',
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
  },
  cardContent: {
    display: 'flex',
    flexDirection: 'column',
    padding: '20px 15px 0 15px',
    textAlign: 'right',
  },
  cardTitle: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    fontSize: '13.3px',
    marginBottom: '35px',
  },
  cardDescriptionNormal: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    fontSize: '20px',
    fontWeight: '700',
  },
  cardDescriptionPositive: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    color: '#70A800',
    fontSize: '20px',
    fontWeight: '700',
  },
  cardDescriptionNegative: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    color: '#B94A48',
    fontSize: '20px',
    fontWeight: '700',
  },
  input: {
    width: '100%',
  },
  paper: {
    padding: '20px',
  },
  popover: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    pointerEvents: 'none',
    marginTop: '15px',
  },
  popperClose: {
    pointerEvents: 'none',
  },
  hidden: {
    // visibility: 'hidden',
    display: 'none',
  },
  color: {
    color: '#3BB3E4',
    position: 'relative',
    top: '-11px',
  },
  infoMessage: {
    fontFamily: '\'Lato\', \'Helvetica\', \'Arial\', sans-serif',
    fontSize: '13.3px',
  },
});

type Props = {
  classes: Object,
  title: string,
  description: string,
  infoMessage: string,
  hasInfo: boolean,
};


class SummaryCard extends React.Component<Props> {
  /**
   * converts the given percent of profit/loss and makes it into number
   * depends if its positive or negative, return appropriate class name
   */
  state = {
    anchorEl: null,
  };

  handlePopoverOpen = (event: SyntheticEvent) => {
    this.setState({ anchorEl: event.target });
  };

  handlePopoverClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const {
      classes,
      title,
      description,
      hasInfo,
      infoMessage,
      // iconColor,
    } = this.props;
    const { anchorEl } = this.state;
    const open = !!anchorEl;
    const totalProfitLoss = () => {
      const result = parseFloat(description.slice(0, description.length - 1));

      if (title === 'Total profit/loss' && result > 0) {
        return classes.cardDescriptionPositive;
      } else if (title === 'Total profit/loss' && result < 0) {
        return classes.cardDescriptionNegative;
      }

      return classes.cardDescriptionNormal;
    };

    return (
      <Card className={classes.cardContainer}>
        <CardHeader
          className={classes.cardHeader}
          avatar={<this.props.icon className={classes.icon} />}
        />
        <CardContent className={classes.cardContent}>
          <Typography component="p" className={classes.cardTitle}>
            {title}
          </Typography>
          <div className={classes.info}>
            <IconButton
              className={hasInfo ? classes.color : classes.hidden}
            >
              <Info
                fill="#3BB3E4"
                onMouseOver={this.handlePopoverOpen}
                onFocus={this.handlePopoverOpen}
                onMouseOut={this.handlePopoverClose}
                onBlur={this.handlePopoverClose}
              />
            </IconButton>
            <Popover
              className={classes.popover}
              classes={{
                paper: classes.paper,
              }}
              open={open}
              anchorEl={anchorEl}
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'center',
              }}
              transformOrigin={{
                vertical: 'top',
                horizontal: 'center',
              }}
              onClose={this.handlePopoverClose}
              disableRestoreFocus
            >
              <Typography className={classes.infoMessage}>{infoMessage}</Typography>
            </Popover>
            <Typography
              type="headline"
              component="h2"
              className={title === 'Total profit/loss' ? totalProfitLoss() : classes.cardDescriptionNormal}
            >
              {description}
            </Typography>
          </div>
        </CardContent>
      </Card>
    );
  }
}

export default withStyles(styles, statsCardStyle)(SummaryCard);
