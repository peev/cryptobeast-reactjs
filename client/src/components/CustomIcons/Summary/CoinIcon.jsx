import React from 'react';
import SvgIcon from 'material-ui/SvgIcon';
import '../icon.css';


const wrapSvgPath = (path, viewBox = '0 0 115 120') => props => (
  <SvgIcon {...props} viewBox={viewBox}>
    {path}
  </SvgIcon>
);

const CoinIcon = (
  <svg viewBox="0 0 401.601 401.6">
    <path d="M116.682 229.329c11.286 0 22.195-.729 32.518-2.086V114.094c-10.322-1.356-21.232-2.085-32.518-2.085C52.241 112.009.001 135.702.001 164.93v11.477c0 29.227 52.24 52.922 116.681 52.922zM116.682 288.411c11.286 0 22.195-.729 32.518-2.084v-33.166c-10.325 1.356-21.229 2.095-32.518 2.095-56.25 0-103.199-18.054-114.227-42.082-1.606 3.5-2.454 7.124-2.454 10.839v11.477c0 29.228 52.24 52.921 116.681 52.921zM149.199 314.823v-2.578c-10.325 1.356-21.229 2.095-32.518 2.095-56.25 0-103.199-18.054-114.227-42.082C.848 275.757 0 279.381 0 283.096v11.477c0 29.229 52.24 52.922 116.681 52.922 12.887 0 25.282-.95 36.873-2.7-2.873-5.877-4.355-12.075-4.355-18.496v-11.476zM284.92 22.379c-64.441 0-116.681 23.693-116.681 52.921v11.477c0 29.228 52.24 52.921 116.681 52.921 64.44 0 116.681-23.693 116.681-52.921V75.3c0-29.228-52.241-52.921-116.681-52.921z" data-original="#000000" className="active-path" data-old_color="#ffffff" fill="#fff" />
    <path d="M284.92 165.626c-56.25 0-103.199-18.053-114.227-42.082-1.606 3.499-2.454 7.123-2.454 10.839v11.477c0 29.228 52.24 52.921 116.681 52.921 64.44 0 116.681-23.693 116.681-52.921v-11.477c0-3.716-.848-7.34-2.454-10.839-11.028 24.029-57.977 42.082-114.227 42.082z" data-original="#000000" className="active-path" data-old_color="#ffffff" fill="#fff" />
    <path d="M284.92 224.71c-56.25 0-103.199-18.054-114.227-42.082-1.606 3.499-2.454 7.123-2.454 10.839v11.477c0 29.229 52.24 52.922 116.681 52.922 64.44 0 116.681-23.693 116.681-52.922v-11.477c0-3.716-.848-7.34-2.454-10.839-11.028 24.029-57.977 42.082-114.227 42.082z" data-original="#000000" className="active-path" data-old_color="#ffffff" fill="#fff" />
    <path d="M284.92 286.983c-56.25 0-103.199-18.054-114.227-42.082-1.606 3.5-2.454 7.123-2.454 10.838v11.478c0 29.228 52.24 52.921 116.681 52.921 64.44 0 116.681-23.693 116.681-52.921v-11.478c0-3.715-.848-7.34-2.454-10.838-11.028 24.027-57.977 42.082-114.227 42.082z" data-original="#000000" className="active-path" data-old_color="#ffffff" fill="#fff" />
    <path d="M284.92 346.066c-56.25 0-103.199-18.053-114.227-42.081-1.606 3.5-2.454 7.125-2.454 10.838V326.3c0 29.228 52.24 52.921 116.681 52.921 64.44 0 116.681-23.693 116.681-52.921v-11.478c0-3.715-.848-7.34-2.454-10.838-11.028 24.028-57.977 42.082-114.227 42.082z" data-original="#000000" className="active-path" data-old_color="#ffffff" fill="#fff" />
  </svg>
);

export default wrapSvgPath(CoinIcon);
