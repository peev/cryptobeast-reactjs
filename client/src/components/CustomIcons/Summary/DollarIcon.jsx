import React from 'react';
import SvgIcon from 'material-ui/SvgIcon';
import '../icon.css';


const wrapSvgPath = (path, viewBox = '0 0 15 20') => props => (
  <SvgIcon {...props} viewBox={viewBox}>
    {path}
  </SvgIcon>
);

const DollarIcon = (
  <svg viewBox="0 0 79.536 79.536">
    <path d="M36.081 60.766V42.145c-6.892-1.662-11.94-4.173-15.159-7.537-3.213-3.366-4.826-7.449-4.826-12.249 0-4.868 1.82-8.951 5.456-12.252 3.643-3.304 8.479-5.209 14.529-5.706V0h7.643v4.401c5.588.557 10.036 2.162 13.334 4.821 3.309 2.651 5.411 6.198 6.333 10.641l-13.344 1.468c-.803-3.5-2.915-5.864-6.323-7.109v17.38c8.436 1.926 14.184 4.422 17.233 7.498 3.065 3.065 4.588 6.996 4.588 11.801 0 5.364-1.927 9.89-5.773 13.561-3.843 3.682-9.202 5.93-16.048 6.758v8.316h-7.643v-8.088C30 70.815 25.062 68.905 21.267 65.7c-3.798-3.2-6.219-7.716-7.275-13.551l13.756-1.248c.562 2.366 1.616 4.422 3.169 6.136 1.542 1.724 3.267 2.967 5.164 3.729zm0-46.682c-2.079.598-3.733 1.61-4.958 3.027-1.225 1.421-1.846 2.99-1.846 4.714 0 1.569.557 3.027 1.688 4.373 1.124 1.359 2.825 2.439 5.111 3.267V14.084h.005zm7.643 47.308c2.631-.409 4.774-1.439 6.422-3.086 1.656-1.657 2.479-3.583 2.479-5.81 0-1.979-.698-3.697-2.076-5.137-1.388-1.43-3.661-2.538-6.825-3.294v17.327z" data-original="#010002" className="active-path" data-old_color="#ffffff" fill="#fff" />
  </svg>
);

export default wrapSvgPath(DollarIcon);
