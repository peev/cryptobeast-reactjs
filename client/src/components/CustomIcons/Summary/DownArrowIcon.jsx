import React from 'react';
import SvgIcon from '@material-ui/core/SvgIcon';
import '../icon.css';


const wrapSvgPath = (path, viewBox = '0 0 15 20') => props => (
  <SvgIcon {...props} viewBox={viewBox}>
    {path}
  </SvgIcon>
);

const DownArrowIcon = (
  // <path
  //   fill="#ca3f58"
  //   id="down-arrow"
  //   data-name="down arrow"
  //   d="M1204.62,788v8.392H1201l7,7.611,7-7.611h-3.62V788h-6.76Z"
  //   transform="translate(-1201 -788)"
  // />

  <svg viewBox="0 0 145.17 72.5">
    <defs />
    <g id="Layer_2" data-name="Layer 2">
      <polygon
        className="cls-9"
        points="145.17 0 72.67 72.5 0 0 145.17 0"
        id="Layer_1-2"

        data-name="Layer 1" />
    </g>
  </svg>
);

export default wrapSvgPath(DownArrowIcon);
