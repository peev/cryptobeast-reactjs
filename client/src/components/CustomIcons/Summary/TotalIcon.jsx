import React from 'react';
import SvgIcon from 'material-ui/SvgIcon';
import '../icon.css';


const wrapSvgPath = (path, viewBox = '0 0 115 120') => props => (
  <SvgIcon {...props} viewBox={viewBox}>
    {path}
  </SvgIcon>
);

const TotalIcon = (
  <svg viewBox="0 0 341.333 341.333">
    <g id="svg_3" fill="#000000" fillOpacity="1">
      <path data-original="#000000" className="active-path" data-old_color="#FBFBFB" fill="#0000" d="M298.667 0h-256v42.667l138.666 128-138.666 128v42.666h256v-64H149.333L256 170.667 149.333 64h149.334z" />
    </g>
  </svg>
);

export default wrapSvgPath(TotalIcon);
