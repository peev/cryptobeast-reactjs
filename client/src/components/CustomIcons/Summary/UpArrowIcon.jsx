import React from 'react';
import SvgIcon from 'material-ui/SvgIcon';
import '../icon.css';


const wrapSvgPath = (path, viewBox = '0 0 15 20') => props => (
  <SvgIcon {...props} viewBox={viewBox} >
    {path}
  </SvgIcon>
);

const UpArrowIcon = (
  // <path
  //   fill="#3ec39d"
  //   id="up-arrow"
  //   data-name="up arrow"
  //   d="M1070.38,804V795.61H1074L1067,788l-7,7.612h3.62V804h6.76Z"
  //   transform="translate(-1060 -788)"
  // />
  <svg viewBox="0 0 145.17 72.5">
    <g id="Layer_2" data-name="Layer 2">
      <polygon
        className="cls-8"
        points="0 72.5 72.5 0 145.17 72.5 0 72.5"
        id="Layer_1-2"
        data-name="Layer 1"
      />
    </g>
  </svg>
);

export default wrapSvgPath(UpArrowIcon);
