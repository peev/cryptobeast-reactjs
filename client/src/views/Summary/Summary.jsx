// @flow
import React from 'react';
import { withStyles, Grid } from 'material-ui';
import { inject, observer } from 'mobx-react';

import summaryStyle from './../../variables/styles/summaryStyle';

import AnalyticsIcon from './../../components/CustomIcons/Summary/AnalyticsIcon';
import TotalIcon from './../../components/CustomIcons/Summary/TotalIcon';
import CoinIcon from './../../components/CustomIcons/Summary/CoinIcon';
import DollarIcon from './../../components/CustomIcons/Summary/DollarIcon';
import AscendantBarsIcon from './../../components/CustomIcons/Summary/AscendantBarsIcon';
import SummaryCard from './../../components/Cards/Summary/SummaryCard';
import AssetBreakdown from './../../components/Cards/Summary/AssetBreakdown';
import SummaryTabs from './../../components/Tabs/SummaryTabs';
import PortfolioSummaryTable from './../../components/CustomTables/PortfolioSummaryTable';

const styles = () => ({
  container: {
    // marginTop: '20px',
  },
  containerHeader: {
    display: 'flex',
    justifyContent: 'space-between',
    margin: '0',
  },
  containerMiddle: {
    margin: '50px 0',
  },
  warningText: {
    marginTop: '35%',
    textAlign: 'center',
  },
  removePaddingBottom: {
    paddingBottom: '0',
  },
});

type Props = {
  classes: Object,
  PortfolioStore: Object,
};

const Summary = inject('PortfolioStore')(observer(({ ...props }: Props) => {
  const { classes, PortfolioStore } = props;

  const handleInfoMessage = () => {
    if (PortfolioStore.summaryTotalInvestmentInUSD === 0 && PortfolioStore.currentPortfolioCostInUSD > 0) {
      return ('Please add new investor or edit your personal investment amount.');
    } else if (PortfolioStore.summaryTotalInvestmentInUSD > 0 && PortfolioStore.currentPortfolioCostInUSD === 0) {
      return ('Please add assets to the portfolio to see total profit/loss.');
    } else {
      return ('Please add new investor or edit your personal investment amount. Please add some assets to your portfolio.');
    }
  };

  return (
    <Grid container>
      <Grid container className={classes.container}>
        <Grid container spacing={8} className={classes.containerHeader}>
          <SummaryCard
            icon={TotalIcon}
            iconColor="gray"
            title="Total number of shares"
            description={PortfolioStore.summaryTotalNumberOfShares}
          />

          <SummaryCard
            icon={AscendantBarsIcon}
            iconColor="gray"
            title="Share price"
            description={PortfolioStore.summaryTotalNumberOfShares !== 0
              ? `$${PortfolioStore.currentPortfolioSharePrice.toFixed(2)}`
              : ''}
            hasInfo={PortfolioStore.summaryTotalNumberOfShares === 0}
            infoMessage="Please add an investment to see your current share price"
          />

          <SummaryCard
            icon={DollarIcon}
            iconColor="gray"
            title="USD equivalent"
            // description={`$${PortfolioStore.currentPortfolioCostInUSD.toFixed(2) || ''}`}
            description={PortfolioStore.currentPortfolioCostInUSD !== 0
              ? `$${PortfolioStore.currentPortfolioCostInUSD.toFixed(2)}`
              : ''}
            hasInfo={PortfolioStore.currentPortfolioCostInUSD === 0}
            infoMessage="Please add assets to your portfolio"
          />

          <SummaryCard
            icon={CoinIcon}
            iconColor="gray"
            title="Total purchase price"
            description={PortfolioStore.summaryTotalInvestmentInUSD !== 0
              ? `$${PortfolioStore.summaryTotalInvestmentInUSD}`
              : ''}
            hasInfo={PortfolioStore.summaryTotalInvestmentInUSD === 0}
            infoMessage="Please add new investor or edit your personal investment amount"
          />

          <SummaryCard
            icon={AnalyticsIcon}
            iconColor="gray"
            title="Total profit/loss"
            description={PortfolioStore.summaryTotalProfitLoss === 0
              ? ''
              : PortfolioStore.summaryTotalProfitLoss > 0
                ? `+${PortfolioStore.summaryTotalProfitLoss}%`
                : `${PortfolioStore.summaryTotalProfitLoss}%`}
            hasInfo={(PortfolioStore.summaryTotalInvestmentInUSD === 0 && PortfolioStore.currentPortfolioCostInUSD > 0)
              || (PortfolioStore.summaryTotalInvestmentInUSD > 0 && PortfolioStore.currentPortfolioCostInUSD === 0)
              || (PortfolioStore.summaryTotalInvestmentInUSD === 0 && PortfolioStore.currentPortfolioCostInUSD === 0)}
            infoMessage={handleInfoMessage()}
          />
        </Grid>

        <Grid container className={classes.containerMiddle}>
          <Grid item xs={6} sm={6} md={6} className={classes.removePaddingBottom}>
            <SummaryTabs />
          </Grid>
          <Grid item xs={6} sm={6} md={6}>
            <AssetBreakdown />
          </Grid>
        </Grid>

        <Grid container className={classes.container}>
          <Grid item xs={12} sm={12} md={12}>
            <PortfolioSummaryTable
              tableHead={[
                { id: 'ticker', numeric: false, disablePadding: false, label: 'Ticker' },
                { id: 'holdings', numeric: false, disablePadding: false, label: 'Holdings' },
                { id: 'priceBTC', numeric: false, disablePadding: false, label: 'Price (BTC)' },
                { id: 'priceUSD', numeric: false, disablePadding: false, label: 'Price (USD)' },
                { id: 'totalUSD', numeric: false, disablePadding: false, label: 'Total Value (USD)' },
                { id: 'assetWeight', numeric: false, disablePadding: false, label: 'Asset Weight' },
                { id: '24Change', numeric: false, disablePadding: false, label: '24H Change' },
                { id: '7Change', numeric: false, disablePadding: false, label: '7D Change' },
              ]}
            />
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
}));


export default withStyles(styles, summaryStyle)(Summary);
