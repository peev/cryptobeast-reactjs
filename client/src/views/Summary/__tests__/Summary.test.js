import React from 'react';
import { shallow } from 'enzyme';
import Summary from './../Summary';

describe('Views/Summary.jsx', () => {
  const summary = shallow(<Summary />);

  it('should render without crashing', () => {
    expect(summary).toMatchSnapshot();
  });
});
