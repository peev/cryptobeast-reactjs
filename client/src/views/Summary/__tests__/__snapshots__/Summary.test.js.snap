// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Summary.jsx should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(inject-Summary-with-PortfolioStore) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "container": "inject-Summary-with-PortfolioStore-container-1",
        "containerHeader": "inject-Summary-with-PortfolioStore-containerHeader-2",
        "containerMiddle": "inject-Summary-with-PortfolioStore-containerMiddle-3",
        "removePaddingBottom": "inject-Summary-with-PortfolioStore-removePaddingBottom-5",
        "warningText": "inject-Summary-with-PortfolioStore-warningText-4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "container": "inject-Summary-with-PortfolioStore-container-1",
          "containerHeader": "inject-Summary-with-PortfolioStore-containerHeader-2",
          "containerMiddle": "inject-Summary-with-PortfolioStore-containerMiddle-3",
          "removePaddingBottom": "inject-Summary-with-PortfolioStore-removePaddingBottom-5",
          "warningText": "inject-Summary-with-PortfolioStore-warningText-4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
